src/LaravelDataTables/Drivers/PropelDataTablesDriver.php

```diff
@@ -248,6 +249,10 @@ public function doFilter()
                 $filterValue = $filter->castValue($rawFilterValue);
                 $relationships = explode('.', $filter->getRelationshipPath());
                 $filterField = array_pop($relationships);
+                if (is_string($filter->getRequestPath()) && array_key_exists($filter->getRequestPath(), $query->getAsColumns()) && $filterValue instanceof Collection) {
+                    $query->addUsingAlias($filter->getRelationshipPath(), $filterValue->all(), Criteria::IN);
+                    continue;
```

Comment:
Always try to avoid using confines in loops, especially large nested loops. Same goes for break. They can be used but should never be relied upon, always try to factor your code so they don't need them where possible. This is just advice, as this should be changed based on the previous comment